### ISSUE TITLE
Add new Cupertino calendar date picker mode

### ISSUE THREAD

**Issue by xster** [flutter/flutter#63693]
**Known Packages**
- https://pub.dev/packages/cupertino_calendar_picker

---

Apple introduced a new full calendar date picker mode in iOS 14 
https://developer.apple.com/design/human-interface-guidelines/ios/controls/pickers/

![date-picker-date_2x-1](https://user-images.githubusercontent.com/156888/90176697-37241f80-dd5e-11ea-9c82-7d2546f7d36f.png)

**Comment by sarp86** [comment]
How long it will take to develop Time picker?

**Comment by IonVillarreal** [comment]
Some alternative ?

**Comment by eli-front** [comment]
update?

**Comment by ebraamboshra1** [comment]
update ?

**Comment by jacobthecoder** [comment]
Update please?

**Comment by SAGARSURI** [comment]
Any update on this?

**Comment by AsjadSiddiqui** [comment]
Update ?

**Comment by vincentderidder** [comment]
Any update?

**Comment by valterh4ck3r** [comment]
any update ?

**Comment by AsjadSiddiqui** [comment]
Any Update?

**Comment by lchwe** [comment]
does flutter ever get ui updates?

**Comment by scotch115** [comment]
A lot of us would really love this UI in our apps...anything in the works devs? 🤞

**Comment by eli-front** [comment]
This is crazy. Does anyone want to work on a PR with me to fix this?

**Comment by valterh4ck3r** [comment]
Have some alternative ?  third library ?  any ?

**Comment by dominikheeb** [comment]
Is there any update on this request? Waiting to use the full datepicker for an ios iPad app. The related PR seems to be months open :S

**Comment by valterh4ck3r** [comment]
@TahaTesser do you publish this component in another package separated ?

publish a plugin with this component, please.

**Comment by jahnli** [comment]
update ？

**Comment by TahaTesser** [comment]
Thanks for the comment everyone, I plan to update the Calendar in the PR with [CalendarDatePicker](https://master-api.flutter.dev/flutter/material/CalendarDatePicker-class.html).adaptive Calendar, it will take advantage of existing logic and tests while adapting to iOS calendar's behavior and style, soon.

**Comment by valterh4ck3r** [comment]
@TahaTesser  thank you so much.

**Comment by reijerteunis** [comment]
Can't wait for it 👯

**Comment by DaniCastroDev** [comment]
Any updates on this? Thanks!

**Comment by stllfe** [comment]
Guys?

**Comment by reijerteunis** [comment]
I didn't wait for it, so I went trying if I could achieve the same with Material and I could.

If you want I can share this code snippets with you.

**Comment by reijerteunis** [comment]
For the ones that want to do the same, please copy the needed from this code:

```
class _DateTimeInput extends StatefulWidget {
  const _DateTimeInput({Key? key, required this.focusNode}) : super(key: key);

  final FocusNode focusNode;

  @override
  State<_DateTimeInput> createState() => _DateTimeInputState();
}

class _DateTimeInputState extends State<_DateTimeInput> {
  late bool showCalendar;
  late bool showTimePicker;

  late DateTime selectedDateTime;

  Timer? _debounce;

  @override
  void initState() {
    super.initState();
    showCalendar = false;
    showTimePicker = false;

    selectedDateTime = DateTime.now();

    context
        .read<GlucoseFormBloc>()
        .add(DateTimeChanged(dateTime: selectedDateTime.toString()));
  }

  @override
  void dispose() {
    _debounce?.cancel();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    var theme = Theme.of(context);
    final customColors = theme.extension<CustomColors>()!;

    final l10n = context.l10n;

    var tag = Localizations.maybeLocaleOf(context)?.toLanguageTag();
    var dateFormatter = DateFormat(DateFormat.YEAR_ABBR_MONTH_DAY, tag);
    var timeFormatter = DateFormat(DateFormat.HOUR_MINUTE, tag);

    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 4.0),
      child: Container(
        decoration: BoxDecoration(
          borderRadius: BorderRadius.circular(8.0),
          border: Border.all(color: const Color(0xFFF4F5FF)),
          boxShadow: [
            BoxShadow(
              color: const Color(0xFFCFD4FF).withOpacity(0.06),
              spreadRadius: 2,
              blurRadius: 2,
              offset: const Offset(0, 1),
            ),
          ],
          color: customColors.white,
        ),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          mainAxisAlignment: MainAxisAlignment.start,
          children: [
            Container(
              height: 47.0,
              padding: const EdgeInsets.only(left: 16.0, right: 8.0),
              child: Row(
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: [
                  Text(l10n.logsDateTime),
                  Row(
                    children: [
                      Padding(
                        padding: const EdgeInsets.symmetric(horizontal: 8.0),
                        child: GestureDetector(
                          child: Container(
                            decoration: BoxDecoration(
                              borderRadius: BorderRadius.circular(4.0),
                              color: customColors.lightBlue,
                            ),
                            padding: const EdgeInsets.symmetric(
                              horizontal: 12.0,
                              vertical: 8.0,
                            ),
                            child: Text(
                              dateFormatter.format(selectedDateTime).toString(),
                              style: showCalendar
                                  ? theme.textTheme.bodyMedium!
                                      .copyWith(color: customColors.red)
                                  : theme.textTheme.bodyMedium,
                            ),
                          ),
                          onTap: () => showCalendar
                              ? setState(() => showCalendar = false)
                              : setState(() {
                                  showCalendar = true;
                                  showTimePicker = false;
                                }),
                        ),
                      ),
                      GestureDetector(
                        child: Container(
                          decoration: BoxDecoration(
                            borderRadius: BorderRadius.circular(4.0),
                            color: customColors.lightBlue,
                          ),
                          padding: const EdgeInsets.symmetric(
                            horizontal: 12.0,
                            vertical: 8.0,
                          ),
                          child: Text(
                            timeFormatter.format(selectedDateTime).toString(),
                            style: showTimePicker
                                ? theme.textTheme.bodyMedium!
                                    .copyWith(color: customColors.red)
                                : theme.textTheme.bodyMedium,
                          ),
                        ),
                        onTap: () => showTimePicker
                            ? setState(() => showTimePicker = false)
                            : setState(() {
                                showCalendar = false;
                                showTimePicker = true;
                              }),
                      ),
                    ],
                  ),
                ],
              ),
            ),
            Visibility(
              visible: showCalendar,
              child: Padding(
                padding:
                    const EdgeInsets.symmetric(horizontal: 16.0, vertical: 8.0),
                child: Theme(
                  data: lightThemeData.copyWith(
                    colorScheme: const ColorScheme.light(
                      primary: Color(0xFF5E72FF),
                      onPrimary: Colors.white,
                      onSurface: Colors.black,
                    ),
                  ),
                  child: CalendarDatePicker(
                    firstDate: DateTime(1900),
                    initialDate: selectedDateTime,
                    lastDate: DateTime.now(),
                    onDateChanged: (DateTime value) {
                      setState(() {
                        selectedDateTime = DateTime(
                          value.year,
                          value.month,
                          value.day,
                          selectedDateTime.hour,
                          selectedDateTime.minute,
                        );

                        context.read<GlucoseFormBloc>().add(DateTimeChanged(
                            dateTime: selectedDateTime.toString()));
                      });
                    },
                  ),
                ),
              ),
            ),
            Visibility(
              visible: showTimePicker,
              child: Padding(
                padding:
                    const EdgeInsets.symmetric(horizontal: 16.0, vertical: 8.0),
                child: SizedBox(
                  height: 224,
                  child: CupertinoTheme(
                    data: const CupertinoThemeData(
                      textTheme: CupertinoTextThemeData(
                        dateTimePickerTextStyle: TextStyle(
                          fontSize: 16,
                          fontWeight: FontWeight.w500,
                          color: Colors.black,
                        ),
                      ),
                    ),
                    child: CupertinoDatePicker(
                      mode: CupertinoDatePickerMode.time,
                      initialDateTime: selectedDateTime,
                      use24hFormat:
                          MediaQuery.of(context).alwaysUse24HourFormat,
                      onDateTimeChanged: (DateTime value) {
                        if (_debounce?.isActive ?? false) _debounce!.cancel();
                        _debounce =
                            Timer(const Duration(milliseconds: 500), () {
                          setState(() {
                            selectedDateTime = DateTime(
                              selectedDateTime.year,
                              selectedDateTime.month,
                              selectedDateTime.day,
                              value.hour,
                              value.minute,
                            );
                            context.read<GlucoseFormBloc>().add(DateTimeChanged(
                                dateTime: selectedDateTime.toString()));
                          });
                        });
                      },
                    ),
                  ),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```

**Comment by alwangk** [comment]
Any update on this?

**Comment by kalismeras61** [comment]
any update ?

**Comment by amadeuszblanik** [comment]
Any update?

**Comment by tneils218** [comment]
Any update?

**Comment by reijerteunis** [comment]
Why are you guys not using the solution I gave. That should be a good starting point.

**Comment by valterh4ck3r** [comment]
> Why are you guys not using the solution I gave. That should be a good starting point.

Can you release this on new package ?

**Comment by TiesPol** [comment]
> Why are you guys not using the solution I gave. That should be a good starting point.

Can you give us a screenshot of your datepicker?

**Comment by atif-waheed-cowlar** [comment]
Any update on this package ?

**Comment by dev-devarsh4** [comment]
Is there any update on this request? Waiting to use the full datepicker for an ios iPad app. The related PR seems to be months open :S

**Comment by MitchellGoodwin** [comment]
Soft update on this one: we're looking into what it would take to do this in the best way possible. Ideally, my goal is to make the Cupertino calendar date picker adaptable with the Material version, while giving the ability to more easily make a custom calendar view. Some blockers I've found are #138270 (which I'm working on) and #48017 (which I'm likely working on next). I can't give an accurate time frame for when this issue will be resolved, as there is likely more refactoring that will have to be done along the way, but I'm optimistic. And the enablement work will broadly benefit Cupertino as a whole along the way.

tl/dr no direct work on this yet, but we're heads down on eliminating issues that are blocking it.

**Comment by jacobthecoder** [comment]
I appreciate the update from January. Any news since then?

**Comment by MitchellGoodwin** [comment]
#138270 was finished. I'm currently working on another big project (#42560), which still has a fair bit of work left. The current plan is to finish #48017 by the end of Q3 of this year, and the Cupertino calendar widget by the end of Q4. Very subject to change.

**Comment by MaherSafadii** [comment]
for those who want a want a work around there's [this package](https://pub.dev/packages/cupertino_calendar_picker) it looks identical to the native one, but it doesn't support an inline calendar, the package creator said he will work on it though.

**Comment by kerberjg** [comment]
@MitchellGoodwin I noticed this also makes use of a collapsible container that seems to be standard in a few places in iOS, I just happened to implement it for #60762, I thought I'd let you know in case you'd want to coordinate on this 🤗 


https://github.com/user-attachments/assets/bccc61d2-5225-4b21-876d-986c2fac50a4

**Comment by Piinks** [comment]
For folks blocked on this, there is a package on pub now: https://pub.dev/packages/cupertino_calendar_picker
